src/worker/core/team/valueChange.ts

```diff
@@ -588,4 +624,53 @@ const valueChange = async (
 	return valuesAdd - valuesRemove;
 };
 
+const getModifiedPickRank = async (
+	tid: number,
+	pidsAdd: number[],
+	pidsRemove: number[],
+) => {
+	const teamSeason = await idb.cache.teamSeasons.indexGet(
+		"teamSeasonsBySeasonTid",
+		[tid, g.get("season")],
+	);
+	const teams = g.get("numActiveTeams");
+	const gp = teamSeason?.gp ?? 0;
+	const seasonFraction = gp / g.get("numGames");
+
+	const players = await idb.cache.players.indexGetAll("playersByTid", tid);
+	const playersAfterTrade = players.filter(p => !pidsRemove.includes(p.pid));
+	for (const pid of pidsAdd) {
+		const p = await idb.cache.players.get(pid);
+		if (p) {
+			playersAfterTrade.push(p);
+		}
+	}
+	const playerRatings = playersAfterTrade.map(p => ({
+		pid: p.pid,
+		value: p.value,
+		ratings: {
+			ovr: p.ratings.at(-1)!.ovr,
+			ovrs: p.ratings.at(-1)!.ovrs,
+			pos: p.ratings.at(-1)!.pos,
+		},
+	}));
+
+	const newTeamOvr = team.ovr(playerRatings);
+	const newTeamOvrRank =
+		newTeamOvr < cache.teamOvrs[cache.teamOvrs.length - 1].ovr
+			? cache.teamOvrs.length - 1
+			: cache.teamOvrs.findIndex(t => t.ovr < newTeamOvr);
+	const newTeamOvrWinp = 0.25 + (0.5 * (teams - 1 - newTeamOvrRank)) / teams;
+	const newWp =
+		gp === 0
+			? newTeamOvrWinp
+			: seasonFraction * ((teamSeason?.won ?? 0) / gp) +
+			  (1 - seasonFraction) * newTeamOvrWinp;
+	const newRank =
+		newWp > cache.wps[cache.wps.length - 1]
+			? cache.wps.length
+			: cache.wps.findIndex(wp => newWp < wp) + 1;
```

Comment:
After fixing the above stuff, one thing to check is to make sure that the rank calculated here is the same as the rank in `cache.estPicks`, for the case where no players are in the trade. Currently it's off by one - like if you have the 4th highest ovr team, `newTeamOvrRank` will be 4 (meaning the 5th best team, since it starts at 0) resulting in a value here of 26 (for 30 teams), when it should be 27 like in `cache.estPicks`.